### PR TITLE
Added required field check in wizard flow

### PR DESCRIPTION
**Pull request by GowthamShanmugam**
BZ: https://bugzilla.redhat.com/show_bug.cgi?id=2236436

### CODE REVIEW

**Comment by GowthamShanmugam** [comment]
[Screencast from 2023-09-11 16-20-43.webm](https://github.com/red-hat-storage/odf-console/assets/15073590/10cab075-1f72-4f4a-b150-7d4572c14807)

File: packages/mco/components/modals/app-manage-policies/assign-policy-view.tsx [hunk 1]
```diff
@@ -89,16 +86,7 @@ export const AssignPolicyView: React.FC<AssignPolicyViewProps> = ({
...
-
-  const onNext = ({ id }: WizardStep) => {
-    if (id) {
-      if (typeof id === 'string') {
-        id = parseInt(id, 10);
-      }
-      setStepIdReached(stepIdReached < id ? id : stepIdReached);
-    }
-  };
+  const [isValidationEnabled, setEnableValidation] = React.useState(false);
```
Comment (SanjalKatiyar) [inline packages/mco/components/modals/app-manage-policies/assign-policy-view.tsx:89]: nits:
```suggestion
  const [isValidationEnabled, setIsValidationEnabled] = React.useState(false);
```

File: packages/mco/components/modals/app-manage-policies/helper/assign-policy-view-footer.tsx [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+  const canJumpToNext = canJumpToNextStep(stepName, dataPolicy, t);
+  const validationError = isValidationEnabled && !canJumpToNext;
+
+  const moveToNextStep = () => {
+    setEnableValidation(true);
+    if (canJumpToNext) {
+      setStepIdReached(stepIdReached <= stepId ? stepId + 1 : stepIdReached);
+      onNext();
+      setEnableValidation(false);
+    }
```
Comment (SanjalKatiyar) [inline packages/mco/components/modals/app-manage-policies/helper/assign-policy-view-footer.tsx:67]:
```suggestion
    if (canJumpToNext) {
      setStepIdReached(stepIdReached <= stepId ? stepId + 1 : stepIdReached);
      onNext();
      setEnableValidation(false);
    } else setEnableValidation(true);
```

File: packages/mco/components/modals/app-manage-policies/helper/pvc-details-wizard-content.tsx [hunk 3]
```diff
@@ -174,9 +183,21 @@ const PairElement: React.FC<PairElementProps> = ({
...
         />
-      </div>
-      <div className="col-xs-5 pairs-list__value-field">
+      </FormGroup>
+      <FormGroup
+        className="col-xs-5 pairs-list__value-field"
+        hasNoPaddingTop
+        isRequired
+        validated={
+          isValidationEnabled && !selectedLabels?.length ? 'error' : 'default'
```
Comment (SanjalKatiyar) [inline packages/mco/components/modals/app-manage-policies/helper/pvc-details-wizard-content.tsx:197]: can we create a small util function for mco ?
`const getValidatedProp = (error: boolean) => error ? 'error' : 'default'`;

File: packages/mco/components/modals/app-manage-policies/helper/assign-policy-view-footer.tsx [hunk 4]
```diff
@@ -0,0 +1,138 @@
...
+  Alert,
+  AlertVariant,
+} from '@patternfly/react-core';
+import { DRPolicyType, DataPolicyType } from '../utils/types';
+import '../../../../style.scss';
+import '../style.scss';
+
+const isPVCSelectorFound = (dataPolicy: DRPolicyType) =>
+  !!dataPolicy?.placementControlInfo?.length &&
+  !!dataPolicy?.placementControlInfo?.every(
```
Comment (SanjalKatiyar) [inline packages/mco/components/modals/app-manage-policies/helper/assign-policy-view-footer.tsx:20]:
```suggestion
  !!dataPolicy.placementControlInfo.every(
```

Comment (SanjalKatiyar) [reply packages/mco/components/modals/app-manage-policies/helper/assign-policy-view-footer.tsx:20]: if `!!dataPolicy?.placementControlInfo?.length` is true then only we will reach to next condition... and if that is happening it means all fields are present...

File: packages/mco/components/modals/app-manage-policies/helper/assign-policy-view-footer.tsx [hunk 5]
```diff
@@ -0,0 +1,138 @@
...
+    (drpc) => !!drpc?.pvcSelector?.length
```
Comment (SanjalKatiyar) [inline packages/mco/components/modals/app-manage-policies/helper/assign-policy-view-footer.tsx:21]:
```suggestion
    (drpc) => !!drpc.pvcSelector?.length
```

**Review by SanjalKatiyar — CHANGES_REQUESTED** [review]

File: packages/mco/components/modals/app-manage-policies/helper/pvc-details-wizard-content.tsx (hunk 3 above)
Comment (SanjalKatiyar) [reply packages/mco/components/modals/app-manage-policies/helper/pvc-details-wizard-content.tsx:197]: PF also have a constant for 'error' & 'default', we can use that directly...

File: packages/mco/components/modals/app-manage-policies/helper/assign-policy-view-footer.tsx (hunk 4 above)
Comment (GowthamShanmugam) [reply packages/mco/components/modals/app-manage-policies/helper/assign-policy-view-footer.tsx:20]: ack

File: packages/mco/components/modals/app-manage-policies/assign-policy-view.tsx [hunk 6]
```diff
@@ -141,7 +128,7 @@ export const AssignPolicyView: React.FC<AssignPolicyViewProps> = ({
     };
     // assign DRPolicy
     const promises = assignPromises(state.policy);
-    Promise.all(promises)
+    await Promise.all(promises)
```
Comment (SanjalKatiyar) [inline packages/mco/components/modals/app-manage-policies/assign-policy-view.tsx:131]: not needed... still not blocking the PR for this small nit...
```suggestion
    Promise.all(promises)
```

**Pull request merged**